doc/content/getting-started/migrating/migrating-from-v2.md

```diff
@@ -19,13 +27,82 @@ For a breakdown of differences between {{% ttnv2 %}} and {{% tts %}}, see the [M
 
 **Finally**: Once you are confident that your end devices are working properly, migrate the rest of your devices and gateways to {{% tts %}}.
 
-## Configure V2 CLI
+## Configure ttn-lw-migrate
+
+End devices and applications can easily be migrated from {{% ttnv2 %}} to {{% tts %}} with the [`ttn-lw-migrate`](https://github.com/TheThingsNetwork/lorawan-stack-migrate) tool. This tool is used for exporting end devices and applications to a [JSON file]({{< ref "getting-started/migrating/device-json.md" >}}) containing their description. This file can later be imported in {{% tts %}} as described in the [Import End Devices in The Things Stack]({{< ref "getting-started/migrating/import-devices.md" >}}) section.
+
+First, configure the environment with the following variables modified according to your setup:
+
+```bash
+$ export TTNV2_APP_ID="my-ttn-app"                    # TTN App ID
+$ export TTNV2_APP_ACCESS_KEY="ttn-account-v2.a..."   # TTN App Access Key (needs `devices` permissions)
+$ export FREQUENCY_PLAN_ID="EU_863_870_TTN"           # Frequency Plan ID for exported devices
+```
+
+See [Frequency Plans]({{< ref src="/reference/frequency-plans" >}}) for the list of frequency plans available on {{% tts %}}. Make sure to specify the correct Frequency Plan ID. For example, the ID `EU_863_870_TTN` corresponds to the **Europe 863-870 MHz (SF9 for RX2 - recommended)** frequency plan.
+
+### Private {{% ttnv2 %}} deployments
 
-For using `ttnctl` with **The Things Network**, follow [these instructions](https://www.thethingsnetwork.org/docs/network/cli/quick-start.html). 
+Private {{% ttnv2 %}} deployments are also supported, and require extra configuration. See `ttn-lw-migrate device --help` for more details. For most cases, it is enough to configure ttn-lw-migrate to use the Discovery Server of your installation, by setting the following environment variables:
+
+```bash
+$ export TTNV2_DISCOVERY_SERVER_ADDRESS="discovery.<tenant>.thethings.industries:1900"
+```
+
+>**Note:**: If the Discovery Server is not using TLS, you will need to use the `--ttnv2.discovery-server-insecure` flag when running the `ttn-lw-migrate` commands below.
+
+## Export End Devices from {{% ttnv2 %}}
+
+In order to export a single device, use the following command. The device with ID `mydevice` will exported and saved to `device.json`.
+
+```bash
+$ ttn-lw-migrate devices --source ttnv2 "mydevice" > devices.json
+```
+
+>**Note:**: Payload formatters are not exported. See [Payload Formatters](https://thethingsstack.io/integrations/payload-formatters/).
+
+>**Warning:**: Active device sessions are exported by default. You can disable this by using the `--ttnv2.with-session=false` flag. It is recommended that you do not export session keys for devices that can instead re-join on The Things Stack.
```

Comment:
- Too many colons here (2x)
- What are "_active_ device sessions"? Probably just sessions
- Why is this not recommended? People can't really go to their devices and power cycle them

Reply:
@neoaggelos i don't have these answers so I'm reassigning you